addons/payment/models/payment_acquirer.py

```diff
@@ -535,6 +539,13 @@ def get_next_reference(self, reference):
             ref_suffix += 1
         return reference
 
+    def _generate_callback_hash(self):
+        self.ensure_one()
+        secret = self.env['ir.config_parameter'].sudo().get_param('database.secret')
+        token = '%s%s%s' % (self.callback_ir_model_id.model, self.callback_res_id, self.callback_method)
+        hm = hmac.new(str(secret), token, hashlib.sha1).hexdigest()
```

Comment:
Ok. I don't think I choose sha1 for a performance matter. More a copy and paste matter. Therefore, I agree with your review Dbo Sir.